assets/js/script.js

```diff
@@ -4,124 +4,80 @@ $(document).ready(function() {
+}
 
-  $(window).resize(function() {
-    var newLetterHeight = $(".letters-wrapper ").height()
-    $("#bagLetter1").css({ display: "block", height: newLetterHeight + "px" })
-    $("#bagLetter2").css({ display: "block", height: newLetterHeight + "px" })
-    bagContainerHeight = $(".main-bag-container").width() * .7
-    $(".main-bag-container").height(bagContainerHeight)
-    bagContainerHeight = $(".main-letters-container").width() * .7
-    $(".main-letters-container").height(bagContainerHeight)
-    letterTabHeight = $(".swatch-row").height() / 2
-    $(".custom-letter-container").height(letterTabHeight)
+$(document).on('input', '.js-input', function(evt) {
+  setState({
+    letters: $(this).val()
   })
+})
+$(document).on('click', '.js-swatch-link', function(evt) {
+  var materials = state.materials.concat([]);
+  materials[state.activeLetter] = $(this).data('color')
 
-  var letter1Choice = "black"
-  var letter2Choice = "black"
-
-  el.letterTabs.on('click', function() {
-    el.letterTabs.removeClass("active")
-    el.lettersWrapper.removeClass('active')
-    $(this).addClass("active")
-    $(this).parent().addClass('active')
+  setState({
+    materials: materials
   })
-
-  el.letterTab1.on('click', function() {
-    el.swatch.removeClass('active')
-    el.swatch.each(function() {
-      if ($(this).data("material-swatch") == letter1Choice) {
-        $(this).addClass("active")
-      }
-    })
+})
+$(document).on('click', '.js-tab', function(evt) {
+  setState({
+    activeLetter: $(this).data('index')
   })
+})
 
-  el.letterTab2.on('click', function() {
-    el.swatch.removeClass('active')
-    el.swatch.each(function() {
-      if ($(this).data("material-swatch") == letter2Choice) {
-        $(this).addClass("active")
-      }
-    })
-  })
+function setState(newState) {
+  state = Object.assign({}, state, newState)
+  render()
+}
 
-  el.input.on("input", function() {
-    var $inputText = $(this).val()
+function updateCustomInfo() {
+  var productDescription = 'Danny-Black'
+  if( state.letters.length > 0 ) {
+    productDescription += ` / ${state.letters[0].toUpperCase()}-${state.materials[0]}`
+  }
+  if( state.letters.length > 1 ) {
+    productDescription += ` / ${state.letters[1].toUpperCase()}-${state.materials[1]}`
+  }
 
-    if ($inputText.length > 0) {
-      $("#bagLetter2").hide()
-      renderLetter(1, $inputText.charAt(0), letter1Choice)
-      if ($inputText.length > 1) {
-        renderLetter(2, $inputText.charAt(1), letter2Choice)
-        $("input").blur()
-      }
-      updateCustomInfo()
-    } else {
-      $("#tab-letter1").html("")
-      $("#tab-letter2").html("")
-      $("#bagLetter1").css({ display: "none" })
-      $("#bagLetter2").css({ display: "none" })
-    }
-  })
+  $("#custombar-custom-info").val(productDescription);
+}
 
-  el.swatch.on('click', function() {
-    el.swatch.removeClass("active")
-    $(this).addClass("active")
-    var text = el.input.val()
+function render() {
```

Comment:
in React, the `render` function actually returns templated html, which makes it easier to follow. here we use jquery to update the DOM in two steps.